File: resume-parser/src/main/java/com/yassir/sdit/resumeparser/parser/models/Resume.java
package com.yassir.sdit.resumeparser.parser.models;

import jakarta.persistence.*;

@Entity
public class Resume {
    public enum State {
        WAITING,
        OCR_QUEUED, OCR_PROGRESS, OCR_DONE, ORC_FAILED,
        PARSE_IN_PROGRESS, PARSE_DONE, PARSE_FAILED,
    }

    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private String objectPath;

    public void setId(Long id) {
        this.id = id;
    }

    public void setObjectPath(String objectPath) {
        this.objectPath = objectPath;
    }

    public void setOcrResult(String ocrResult) {
        this.ocrResult = ocrResult;
    }

    public void setParseResult(String parseResult) {
        this.parseResult = parseResult;
    }

    public void setState(State state) {
        this.state = state;
    }

    private String ocrResult;

    private String parseResult;

    public String getObjectPath() {
        return objectPath;
    }

    public Long getId() {
        return id;
    }

    public String getOcrResult() {
        return ocrResult;
    }

    public String getParseResult() {
        return parseResult;
    }

    public State getState() {
        return state;
    }

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private State state;
}

File: sdt-resume-ocr-main/src/resume/resume.py
from enum import Enum


class Resume:
    class State(Enum):
        WAITING = 'WAITING'
        OCR_QUEUED = 'OCR_QUEUED'
        OCR_IN_PROGRESS = 'OCR_IN_PROGRESS'
        OCR_DONE = 'OCR_DONE'
        OCR_FAILED = 'OCR_FAILED'
        PARSE_IN_PROGRESS = 'PARSE_IN_PROGRESS'
        PARSE_DONE = 'PARSE_DONE'
        PARSE_FAILED = 'PARSE_FAILED'

    def __init__(self, resume_id: int, object_path: str, ocr_result: str, parse_result: str, state: State):
        self.id = resume_id
        self.object_path = object_path
        self.ocr_result = ocr_result
        self.parse_result = parse_result
        self.state = state
